src/main/java/it/gov/pagopa/payhub/activities/service/ingestionflow/debtpositiontype/DebtPositionTypeProcessingService.java

```diff
@@ -0,0 +1,130 @@
+
+  private final DebtPositionTypeMapper debtPositionTypeMapper;
+  private final DebtPositionTypeService debtPositionTypeService;
+  private final OrganizationService organizationService;
+
+
+  public DebtPositionTypeProcessingService(
+      DebtPositionTypeMapper debtPositionTypeMapper,
+      DebtPositionTypeErrorsArchiverService debtPositionTypeErrorsArchiverService,
+      DebtPositionTypeService debtPositionTypeService, OrganizationService organizationService) {
+    super(debtPositionTypeErrorsArchiverService);
+    this.debtPositionTypeMapper = debtPositionTypeMapper;
+    this.debtPositionTypeService = debtPositionTypeService;
+    this.organizationService = organizationService;
+  }
+
+
+  public DebtPositionTypeIngestionFlowFileResult processDebtPositionType(
+      Iterator<DebtPositionTypeIngestionFlowFileDTO> iterator,
+      List<CsvException> readerException,
+      IngestionFlowFile ingestionFlowFile, Path workingDirectory) {
+
+    List<DebtPositionTypeErrorDTO> errorList = new ArrayList<>();
+    DebtPositionTypeIngestionFlowFileResult ingestionFlowFileResult = new DebtPositionTypeIngestionFlowFileResult();
+    ingestionFlowFileResult.setDebtPositionTypeCodeList(new ArrayList<>());
+
+    Optional<Organization> organizationBroker = organizationService.getOrganizationById(
+        ingestionFlowFile.getOrganizationId());
+    Organization organization = organizationBroker.orElse(null);
+    Long brokerId = organization != null ? organization.getBrokerId() : null;
+    if (brokerId == null) {
+      log.error("Broker for organization id {} not found", ingestionFlowFile.getOrganizationId());
+      ingestionFlowFileResult.setErrorDescription("Broker not found");
+      return ingestionFlowFileResult;
+    }
+    ingestionFlowFileResult.setBrokerId(brokerId);
+    ingestionFlowFileResult.setBrokerFiscalCode(organization.getOrgFiscalCode());
+
+    process(iterator, readerException, ingestionFlowFileResult, ingestionFlowFile, errorList,
+        workingDirectory);
+    return ingestionFlowFileResult;
+  }
+
+  @Override
+  protected boolean consumeRow(long lineNumber,
+      DebtPositionTypeIngestionFlowFileDTO debtPositionTypeDTO,
+      DebtPositionTypeIngestionFlowFileResult ingestionFlowFileResult,
+      List<DebtPositionTypeErrorDTO> errorList, IngestionFlowFile ingestionFlowFile) {
+    try {
+      CollectionModelDebtPositionType existingDebtPosType = debtPositionTypeService.getByMainFields(
+          debtPositionTypeDTO.getDebtPositionTypeCode(),
+          ingestionFlowFileResult.getBrokerId(),
+          debtPositionTypeDTO.getOrgType(),
+          debtPositionTypeDTO.getMacroArea(),
+          debtPositionTypeDTO.getServiceType(),
+          debtPositionTypeDTO.getCollectingReason(),
+          debtPositionTypeDTO.getTaxonomyCode());
+
+      List<DebtPositionType> debtPositionTypeList = new ArrayList<>();
```

Comment:
don't create object if not necessary
```suggestion
      List<DebtPositionType> debtPositionTypeList = null;
```